### PR TITLE
Added Failure step

### PR DESCRIPTION
**Pull request by eyaliyahu**

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -290,3 +290,39 @@ runs:
...
+      shell: bash
+
+    - name: Notify Jit on Failure
+      if: failure()
+      uses: actions/github-script@v6
+      with:
+        script: |
+          console.log(`Notifying Jit of failure`);
+          const headers = {
+            'Authorization': `Bearer ${{ steps.oidc-auth.outputs.jit_token }}`,
```
Comment (avichayJit) [inline action.yml:306]: we won't always have oidc token
we should use `${{ fromJSON(github.event.inputs.client_payload).jit_jwt_token`

File: action.yml [hunk 2]
```diff
@@ -290,3 +290,39 @@ runs:
...
+            'Content-Type': 'application/json'
+          };
+          console.log(`Headers: ${JSON.stringify(headers)}`);
+          const body = {
+            execution_id: "${{ fromJSON(github.event.inputs.client_payload).payload.execution_id }}",
+            jit_event_id: "${{ fromJSON(github.event.inputs.client_payload).payload.jit_event_id }}",
+            status: "failed"
```
Comment (avichayJit) [inline action.yml:313]: no need to add `status`

File: action.yml [hunk 3]
```diff
@@ -290,3 +290,39 @@ runs:
...
+          };
+          console.log(`Notifying Jit of failure with body: ${JSON.stringify(body)}`);
+          
+          (async () => {
+            const responseTemp = await github.request('GET https://api.ipify.org?format=json');
+            const ip = responseTemp.data.ip;
+            
+            const response = await github.request('POST https://api.jit.io/executions/complete', {
```
Comment (avichayJit) [inline action.yml:321]: this is not the api to call.
`${{ fromJSON(github.event.inputs.client_payload).payload.jit_base_api }}/github/workflow-failed }}`

File: action.yml [hunk 4]
```diff
@@ -290,3 +290,39 @@ runs:
...
+          const body = {
+            execution_id: "${{ fromJSON(github.event.inputs.client_payload).payload.execution_id }}",
+            jit_event_id: "${{ fromJSON(github.event.inputs.client_payload).payload.jit_event_id }}",
+            status: "failed"
+          };
+          console.log(`Notifying Jit of failure with body: ${JSON.stringify(body)}`);
+          
+          (async () => {
+            const responseTemp = await github.request('GET https://api.ipify.org?format=json');
+            const ip = responseTemp.data.ip;
```
Comment (avichayJit) [inline action.yml:319]: what is this?

File: action.yml [hunk 5]
```diff
@@ -290,3 +290,39 @@ runs:
...
         # this field would set the original repo command mount command (would be mounted always in the old flow) and only if checkout = true in the new flow
         mount_original_repo_command: ${{ fromJSON('["","-v $(pwd)/code:/code"]')[((!(fromJSON(github.event.inputs.client_payload).payload.runner_setup) || (fromJSON(github.event.inputs.client_payload).payload.runner_setup.checkout == true)))] }}
+
+    - name: Temp Raise Error
+      if: always()
+      run: exit 1
+      shell: bash
+
+    - name: Notify Jit on Failure
+      if: failure()
```
Comment (avichayJit) [inline action.yml:300]: we shouldn't fall here from `Run The Action` (where we call `docker run`)

**Review by yardensade — APPROVED** [review]

**Pull request merged**